src/FakeItEasy/Core/InterceptedFakeObjectCall.cs

```diff
@@ -0,0 +1,54 @@
+namespace FakeItEasy.Core
+{
+    using System.Reflection;
+    using FakeItEasy.Configuration;
+
+    internal abstract class InterceptedFakeObjectCall : IInterceptedFakeObjectCall

```

Comment:
Maybe I'm missing something, but why is this necessary?

Reply:
It allowed me to return `IOrderedCompletedFakeObjectCall` from `AsReadOnly` when called internally, and only `ICompletedFakeObjectCall` when `AsReadOnly` is invoked through the `IInterceptedFakeObjectCall` interface (that is, by clients).